server/container/guice/guice-common/src/main/java/org/apache/james/utils/FileConfigurationProvider.java

```diff
@@ -58,8 +58,11 @@ public HierarchicalConfiguration getConfiguration(String component) throws Confi
         Preconditions.checkNotNull(component);
         List<String> configPathParts = Splitter.on(".").splitToList(component);
         Preconditions.checkArgument(!configPathParts.isEmpty());
-        HierarchicalConfiguration config = getConfig(retrieveConfigInputStream(configPathParts.get(0)));
-        return selectHierarchicalConfigPart(config, Iterables.skip(configPathParts, 1));
+        return getConfiguration(retrieveConfigInputStream(configPathParts.get(0)), Iterables.skip(configPathParts, 1));
+    }
+
+    public HierarchicalConfiguration getConfiguration(InputStream inputStream, Iterable<String> configsPathParts) throws ConfigurationException {
```

Comment:
Yes public.... I am sure you don't want me to write a Hierarchical configuration by hand. And loading a hierhachical configuration from astream looks like a task for FileConfigurationProvider...